app/js/enhancedDate.js

```diff
@@ -1,5 +1,136 @@
 /* 
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
 
+var enhancedDate = (function () {
+	
+	"use strict";
+
+	var myDate = null;
+	var dayOfWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
+	var month = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+
+	function isDateSet() {
+
+		if (myDate === null) {
+
+			setDefaultDate();
+			
+		} 
+	}
+
+	function setDefaultDate() {
+
+		myDate = new Date();
+	}
+
+	return {
+		
+		/* Stores the passed-in date. 
+		 * This function can take milliseconds after epoch 
+		 * or a Date object. Do some type-checking to make 
+		 * sure you have a valid type. If no value is passed, 
+		 * default your internally stored date to now.
+		 */
+		setDate: function ( date ) {
+
+			if ( typeof date === 'number' ){
+
+				myDate = new Date(date);
+
+			} else if ( date instanceof Date ) {
+
+				myDate = date;
+
+			} else if ( date === undefined ) {
+
+				//setDefaultDate();
+			myDate = new Date();
+			} else {
+			
+				console.log("ERROR: A type other than a Number or Date was provided");
+			
+			}
+		},
+
+		/* Returns either milliseconds after epoch by default 
+		 * or a Date object if true is passed as an argument.
+		 */ 
+		getDate: function ( date ) {
+			
+			isDateSet();
+
+			if ( date === true ){
+
+				return myDate;
+
+			} else {
+				
+				return myDate.getTime();
+
+			}
+		},
+
+		// Returns the full day name as a string, e.g. "Monday", "Tuesday", etc.
+		getDayName: function () {
+			
+			isDateSet();
+
+			return dayOfWeek[ myDate.getDay() ];
+
+		},
+
+		// Return the full month name as a string, e.g. "January", "February", etc.
+		getMonthName: function () {
+			
+			isDateSet();
+
+			return month[ myDate.getMonth() ];
+
+		},
+
+		// Return boolean true if date is in the future (from when method is called)
+		isFuture: function () {
+			
+			isDateSet();
+
+			var dateNow = Date.now();
+
+			if ( myDate.getTime() > dateNow ) {
+
+				return true;
+
+			} else {
+
+				return false;
+
+			}
+		},
+
+		// Return boolean true if the stored date is "today".
+		isToday: function () {
+			
+			isDateSet();
+
+			var dateNow = new Date();
+			var dayNow = dateNow.getDay();
```

Comment:
Instead of dateNow.getDay() on 117, just dateNow.getDate(). Not to be confused with the same method name on enhancedDate - for reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getDate